src/net_processing.cpp

```diff
@@ -1373,7 +1375,11 @@ static void RelayAddress(const CAddress& addr, bool fReachable, const CConnman&
     // Use deterministic randomness to send to the same nodes for 24 hours
     // at a time so the m_addr_knowns of the chosen nodes prevent repeats
     uint64_t hashAddr = addr.GetHash();
-    const CSipHasher hasher = connman.GetDeterministicRandomizer(RANDOMIZER_ID_ADDRESS_RELAY).Write(hashAddr << 32).Write((GetTime() + hashAddr) / (24 * 60 * 60));
+
+    std::chrono::seconds rotate_interval = std::chrono::duration_cast<std::chrono::seconds>(ROTATE_ADDR_RELAY_DEST_INTERVAL);
```

Comment:
Is this duration_cast needed? I think conversion should be automatic.

Reply:
You are right. I spent 30 minutes fighting with this code, and couldn't come up with something better.

But after you messaged I tried once more and it worked. Will force push in a second.